### PR TITLE
feat(subOrgs): get org subOrgs hierarchy

### PR DESCRIPTION
**Pull request by dwojno**

### CODE REVIEW

**Review by kim-energyweb — APPROVED** [review]
LGTM

File: src/organization/organization.service.ts [hunk 1]
```diff
@@ -299,4 +300,58 @@ export class OrganizationService {
...
+          subOrgs: await this.getDefinitionForOrgs(org.subOrgs),
+        })),
+      )
+    );
+  }
+
+  public async getOrganizationNestedSubOrgs(namespace: string) {
+    const res = await this.dgraph.query(`
+    {
+      orgs(func: eq(namespace, "${namespace}")) @filter(type(Org)) @recurse(depth: 20, loop: true) {
```
Comment (jrhender) [inline src/organization/organization.service.ts:340]: Cool, I didn't know about this `@recurse` functionality. Why the need to loop? I would think there wouldn't be loops in the organization hierarchies

File: src/organization/organization.service.ts [hunk 2]
```diff
@@ -299,4 +300,58 @@ export class OrganizationService {
...
+    return definition;
+  }
+
+  private getDefinitionForOrgs(orgs?: Organization[]) {
+    return (
+      orgs &&
+      Promise.all(
+        orgs.map(async org => ({
+          ...org,
+          definition: await this.getOrgDefinitionByUid(org.definition.uid),
```
Comment (jrhender) [inline src/organization/organization.service.ts:330]: Why do you need to get the OrgDefinition again? It seems like a bunch of the data (ie. uid, name, owner, namespace, definition) is being retrieved already in `getOrganizationNestedSubOrgs`?

File: src/organization/organization.service.ts [hunk 3]
```diff
@@ -299,4 +300,58 @@ export class OrganizationService {
     };
     return json?.Data?.[0]?.['subOrgs'] || [];
   }
+
+  private async getOrgDefinitionByUid(uid: string) {
```
Comment (jrhender) [inline src/organization/organization.service.ts:304]: Minor thing, but would be nice to have private methods after public in the file

**Review by jrhender — APPROVED** [review]
Left a few comments/questions, but looks good. The dgraph syntax is tricky! 😆

File: src/organization/organization.service.ts (hunk 2 above)
Comment (dwojno) [reply src/organization/organization.service.ts:330]: In the @recurse function we can get only top level props, so the definition looks like this { uuid: string } and we need to get the rest manually :)

File: src/organization/organization.service.ts (hunk 3 above)
Comment (dwojno) [reply src/organization/organization.service.ts:304]: 👍

File: src/organization/organization.service.ts (hunk 1 above)
Comment (dwojno) [reply src/organization/organization.service.ts:340]: If the loop would be `false` the dgraph will stop looking when found the same node as earlier :) so this is just to be sure to find all nested suborgs in whatever configuration :)

**Pull request merged**